lib/containers/common.pm

```diff
@@ -27,16 +27,13 @@ sub is_unreleased_sle {
 
 sub activate_containers_module {
     my $json = Mojo::JSON::decode_json(script_output_retry('SUSEConnect -s', timeout => 240, retry => 3, delay => 60));
-    my ($not_registered, $version, $arch);
+    my ($running_version, $sp, $host_distri) = get_os_release;
     foreach (@$json) {
-        if ($_->{identifier} =~ 'sle-module-containers' && $_->{status} =~ 'Not Registered') {
-            $not_registered = 1;
-            $version = $_->{version};
-            $arch = $_->{arch};
+        if ($_->{identifier} =~ 'sle-module-containers' && $_->{status} =~ '^Registered') {
+            $running_version eq '12' ? add_suseconnect_product("sle-module-containers", 12) : add_suseconnect_product("sle-module-containers");
```

Comment:
looks like `add_suseconnect_product` is already doing this magic:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/registration.pm#L183

Reply:
I'll try.